### PR TITLE
pass DOTNET_ROOT in tox.ini for non-standard SDK locations

### PR DESCRIPTION
**Pull request by rkm**

### CODE REVIEW

File: pre_commit/languages/dotnet.py [hunk 1]
```diff
@@ -24,6 +24,7 @@
 def get_env_patch(venv: str) -> PatchesT:
     return (
         ('PATH', (os.path.join(venv, BIN_DIR), os.pathsep, Var('PATH'))),
+        ('DOTNET_ROOT', os.environ.get('DOTNET_ROOT', '')),
```
[the remark this answers is not in the document]
Comment (rkm) [reply pre_commit/languages/dotnet.py:27]: thanks for checking -- the tox.ini passenv addition was all I needed after all. Is that still valid as a change on its own in this PR?

[the remark this answers is not in the document]
Comment (asottile) [reply pre_commit/languages/dotnet.py:27]: this line should be removed -- the only side-effect it _could_ have is setting `DOTNET_ROOT` to the empty string which probably isn't expected

**Comment by asottile** [comment]
hmm, the dotnet tests are still failing on linux -- is that expected?

**Comment by asottile** [comment]
ah I think I figured it out -- https://github.com/pre-commit/pre-commit/pull/2634/commits/46c64efd9d8da13e20292f26429d228e4f76958f

**Comment by asottile** [comment]
ugh I goofed by trying these out and then merging the branch they were based on -- can you reopen these (it seems I can't fix them) @rkm

**Pull request closed without merging**